File: V1.1/INTEL-SE/src/main/java/com/intelse/config/ConfigManager.java
package com.intelse.config;

import org.yaml.snakeyaml.Yaml;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileWriter;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.nio.file.attribute.PosixFilePermission; // Added import
import java.util.HashMap;
import java.util.Map;
import java.util.Set; // Added import
import java.util.HashSet; // Added import

public class ConfigManager {
    private static final String CONFIG_PATH = "/home/lxcxjxhx/PROJECT/INTEL-SE/config/settings.yaml";
    private static ConfigManager instance;
    private Map<String, Object> config;

    private ConfigManager() {
        loadConfig();
    }

    public static ConfigManager getInstance() {
        if (instance == null) {
            instance = new ConfigManager();
        }
        return instance;
    }

    private void loadConfig() {
        try {
            File file = new File(CONFIG_PATH);
            if (!file.exists()) {
                config = new HashMap<>();
                config.put("docDir", "/home/lxcxjxhx/PROJECT/INTEL-SE/docs");
                config.put("aiModel", "deepseek");
                config.put("targetIp", "");
                saveConfig();
            } else {
                try (FileInputStream fis = new FileInputStream(file)) {
                    config = new Yaml().load(fis);
                }
            }
        } catch (Exception e) {
            config = new HashMap<>();
        }
    }

    private void saveConfig() {
        try (FileWriter writer = new FileWriter(CONFIG_PATH)) {
            new Yaml().dump(config, writer);
            // Set permissions only on POSIX systems
            if (System.getProperty("os.name").toLowerCase().contains("linux")) {
                Set<PosixFilePermission> perms = new HashSet<>();
                perms.add(PosixFilePermission.OWNER_READ);
                perms.add(PosixFilePermission.OWNER_WRITE);
                perms.add(PosixFilePermission.GROUP_READ);
                perms.add(PosixFilePermission.GROUP_WRITE);
                perms.add(PosixFilePermission.OTHERS_READ);
                Files.setPosixFilePermissions(Paths.get(CONFIG_PATH), perms);
            }
        } catch (Exception e) {
            System.err.println("Config write error: " + e.getMessage());
        }
    }

    public String getDocDir() {
        return (String) config.getOrDefault("docDir", "/home/lxcxjxhx/PROJECT/INTEL-SE/docs");
    }

    public String getAIModel() {
        return (String) config.getOrDefault("aiModel", "deepseek");
    }

    public String getTargetIp() {
        return (String) config.getOrDefault("targetIp", "");
    }

    public void setTargetIp(String ip) {
        config.put("targetIp", ip);
        saveConfig();
    }
}

File: HOS-MANAGER-ALL/INTEL-SE/src/config_manager.py
import yaml
import os

class ConfigManager:
    def __init__(self, config_path):
        self.config_path = config_path
        self.config = self.load_config()

    def load_config(self):
        try:
            with open(self.config_path, "r") as f:
                config = yaml.safe_load(f) or {}
                return config
        except FileNotFoundError:
            return {
                "model": "grok",
                "doc_path": "/home/lxcxjxhx/PROJECT/INTEL-SE/docs",
                "api_key": "",
                "api_endpoint": "https://api.x.ai/v1",
                "tabs": {},
                "enabled_tools": []
            }
        except Exception as e:
            print(f"加载配置错误：{e}")
            return {}

    def save_config(self, config):
        try:
            current_config = self.load_config()
            current_config.update(config)
            with open(self.config_path, "w") as f:
                yaml.dump(current_config, f, allow_unicode=True)
            os.chmod(self.config_path, 0o664)
            self.config = current_config
        except Exception as e:
            print(f"保存配置错误：{e}")

    def get_config(self):
        return self.config
